It is recommended to indicate a format for properties with numeric type.
Defining a format improves interoperability and avoids ambiguities when consuming or generating clients from the API contract.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

```
swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /invoices:
    get:
      responses:
        200:
          description: An invoice.
          schema:
            type: object
            properties:
              productId:
                type: integer
              quantity:
                type: number
              amount:
                type: number
```

Compliant Solution (OpenAPI 2)
------------------------------

```
swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /invoices:
    get:
      responses:
        200:
          description: An invoice.
          schema:
            type: object
            properties:
              productId:
                type: integer
                format: int32
              quantity:
                type: integer
                format: int32
              amount:
                type: number
                format: double
```

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

```
openapi: 3.0.0
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /invoices:
    get:
      responses:
        200:
          description: An invoice.
          content:
            application/json:
              schema:
                type: object
                properties:
                  productId:
                    type: integer
                  price:
                    type: number
```

Compliant Solution (OpenAPI 3)
------------------------------

```
openapi: 3.0.0
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /invoices:
    get:
      responses:
        200:
          description: An invoice.
          content:
            application/json:
              schema:
                type: object
                properties:
                  productId:
                    type: integer
                    format: int32
                  price:
                    type: number
                    format: double
```