The $ref of a response must end with a corresponding suffix

Noncompliant Code Example (OpenAPI 2)
---------------------------------------------------------------------------------------

    swagger: '2.0'
    info:
      title: Sample API
      version: 1.0.0
    paths:
      /sample:
        get:
          responses:
            '200':
              description: Successful response
              schema:
                $ref: '#/definitions/InvalidRef' # Noncompliant {{OAR090: The $ref of a response must end with the suffix Response}}
    definitions:
      InvalidRef:
        type: object
        properties:
          message:
            type: string

Compliant Solution (OpenAPI 2)
---------------------------------------------------------------------------------------

    swagger: '2.0'
    info:
      title: Sample API
      version: 1.0.0
    paths:
      /sample:
        get:
          responses:
            '200':
              description: Successful response
              schema:
                $ref: '#/definitions/SuccessResponse'
    definitions:
      SuccessResponse:
        type: object
        properties:
          message:
            type: string

Noncompliant Code Example (OpenAPI 3)
---------------------------------------------------------------------------------------

    openapi: 3.0.0
    info:
      title: Sample API
      version: 1.0.0
    paths:
      /sample:
        get:
          responses:
            '200':
              $ref: '#/components/responses/InvalidRef' # Noncompliant {{OAR090: The $ref of a response must end with the suffix Response}}
    components:
      responses:
        InvalidRef:
          description: Successful response

Compliant Solution (OpenAPI 3)
---------------------------------------------------------------------------------------

    openapi: 3.0.0
    info:
      title: Sample API
      version: 1.0.0
    paths:
      /sample:
        get:
          responses:
            '200':
              $ref: '#/components/responses/SuccessResponse'
    components:
      responses:
        SuccessResponse:
          description: Successful response