All parameters in query must be defined as optional

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

  swagger: "2.0"
  info:
    title: Swagger Petstore
    version: 1.0.0
  paths:
    /pets:
      get:
        parameters:
          - in: query
            name: $expand
            type: array
            required: true \# Noncompliant {{OAR060: All parameters in query must be defined as optional}}
            items:
              type: string
        responses:
          200:
            description: Ok

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /pets:
    get:
      parameters:
        - in: query
          name: $expand
          type: array
          required: false
          items:
            type: string
      responses:
        200:
          description: Ok

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

  openapi: 3.0.1
  info:
    title: Swagger Petstore
    version: 1.0.0
  paths:
    /pets:
      get:
        parameters:
          - in: query
            name: $orderby
            required: true \# Noncompliant {{OAR060: All parameters in query must be defined as optional}}
            schema:
              type: array
              items:
                type: string
        responses:
          200:
            description: Ok

Compliant Solution (OpenAPI 3)
------------------------------

openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /pets:
    get:
      parameters:
        - in: query
          name: $orderby
          required: false
          schema:
            type: array
            items:
              type: string
      responses:
        200:
          description: Ok