All parameters in query must be snake\_case

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

    swagger: "2.0"
    info:
      version: "1.0.0"
      title: "API que no cumple"
    paths:
      /usuarios:
        get:
          parameters:
            - name: username
              in: query
              type: string
            - name: userFirstName \# Noncompliant {{OAR077: All parameters in query must be snake\_case}}
              in: query
              type: string

Compliant Solution (OpenAPI 2)
------------------------------

    swagger: "2.0"
    info:
      version: "1.0.0"
      title: "API que cumple"
    paths:
      /usuarios:
        get:
          parameters:
            - name: username
              in: query
              type: string
            - name: user\_first\_name
              in: query
              type: string

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

    openapi: "3.0.0"
    info:
      version: "1.0.0"
      title: "API que no cumple"
    paths:
      /usuarios:
        get:
          parameters:
            - name: username
              in: query
              schema:
                type: string
            - name: userFirstName \# Noncompliant {{OAR077: All parameters in query must be snake\_case}}
              in: query
              schema:
                type: string

Compliant Solution (OpenAPI 3)
------------------------------

    openapi: "3.0.0"
    info:
      version: "1.0.0"
      title: "API que cumple"
    paths:
      /usuarios:
        get:
          parameters:
            - name: username
              in: query
              schema:
                type: string
            - name: user\_first\_name
              in: query
              schema:
                type: string