You should categorize the operations of your contract with tags. Categorizing operations helps your users understand the broad organization of the API and highlights the type of manipulated resources.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /pets:
    get: \# Noncompliant {{OAR046: Associate a tag to this operation}}
      responses:
        200:
          description: OK
    post: \# Noncompliant {{OAR046: Associate a tag to this operation}}
      responses:
        default:
          description: the default response

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
tags:
- name: used-tag
  description: a tag referenced in the operations
- name: other-tag
  description: a tag referenced in the operations
paths:
  /pets:
    get:
      tags:
      - used-tag
      responses:
        200:
          description: OK
    post:
      tags:
      - other-tag
      responses:
        default:
          description: the default response

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: 3.0.1
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /pets:
    get: \# Noncompliant {{OAR046: Associate a tag to this operation}}
      responses:
        200:
          description: OK
    post: \# Noncompliant {{OAR046: Associate a tag to this operation}}
      responses:
        default:
          description: the default response

Compliant Solution (OpenAPI 3)
------------------------------

openapi: 3.0.1
info:
  version: 1.0.0
  title: Swagger Petstore
tags:
- name: used-tag
  description: a tag referenced in the operations
- name: other-tag
  description: a tag referenced in the operations
paths:
  /pets:
    get:
      tags:
      - used-tag
      responses:
        200:
          description: OK
    post:
      tags:
      - other-tag
      responses:
        default:
          description: the default response