RequestBody and Responses schema property names must be compliant with the PascalCase naming convention

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

  swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /user:
    post:
      parameters:
        - in: body
          name: body
          required: true
          schema:
            type: object
            properties:
              first\_name: \# Noncompliant {{OAR068: RequestBody and Responses schema property names must be compliant with the PascalCase naming convention}}
                type: string
      responses:
        '200':
          description: Success
          schema:
            type: object
            properties:
              user\_id: \# Noncompliant {{OAR068: RequestBody and Responses schema property names must be compliant with the PascalCase naming convention}}
                type: integer
  

Compliant Solution (OpenAPI 2)
------------------------------

  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /user:
      post:
        parameters:
          - in: body
            name: body
            required: true
            schema:
              type: object
              properties:
                FirstName:
                  type: string
                LastName:
                  type: string
        responses:
          '200':
            description: Success
            schema:
              type: object
              properties:
                UserId:
                  type: integer
                UserName:
                  type: string

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

  openapi: 3.0.0
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /user:
      post:
        requestBody:
          content:
            application/json:
              schema:
                type: object
                properties:
                  first\_name: \# Noncompliant {{OAR068: RequestBody and Responses schema property names must be compliant with the PascalCase naming convention}}
                    type: string
                  last\_name: \# Noncompliant {{OAR068: RequestBody and Responses schema property names must be compliant with the PascalCase naming convention}}
                    type: string
        responses:
          '200':
            description: Success
            content:
              application/json:
                schema:
                  type: object
                  properties:
                    user\_id: \# Noncompliant {{OAR068: RequestBody and Responses schema property names must be compliant with the PascalCase naming convention}}
                      type: integer
                    user\_name: \# Noncompliant {{OAR068: RequestBody and Responses schema property names must be compliant with the PascalCase naming convention}}
                      type: string

Compliant Solution (OpenAPI 3)
------------------------------

  openapi: 3.0.0
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /user:
      post:
        requestBody:
          content:
            application/json:
              schema:
                type: object
                properties:
                  LastName:
                    type: string
        responses:
          '200':
            description: Success
            content:
              application/json:
                schema:
                  type: object
                  properties:
                    UserName:
                      type: string