The string properties of the specified parameters must define a byte or binary format

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

### In case "product, line and price" are the fields to wich the rule is applied

  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /invoices:
      get:
        responses:
          200:
            description: A invoice.
            schema:
              type: object
              properties:
                product: \# Noncompliant {{OAR082: The string properties of the specified parameters must define a byte or binary format}}
                  type: string 
                  format: int128
                line: \# Noncompliant {{OAR082: The string properties of the specified parameters must define a byte or binary format}}
                  type: string 
                lapiz:
                  type: string
                price:  \# Noncompliant {{OAR082: The string properties of the specified parameters must define a byte or binary format}}
                  type: string
                  format: byte

Compliant Solution (OpenAPI 2)
------------------------------

  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /invoices:
      get:
        responses:
          200:
            description: A invoice.
            schema:
              type: object
              properties:
                goma:  
                  type: string 
                  format: byte
                linea: 
                  type: string 
                  format: byte
                lapiz:
                  type: string
                  format: byte
                precio:  
                  type: string
                  format: byte

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

  openapi: "3.0.0"
  info:
    version: "1.0.0"
    title: "Swagger Petstore"
  paths:
    /invoices:
      get:
        responses:
          '200':
            description: A invoice.
            content: 
              application/json:
                schema:
                  type: object
                  properties:
                    product: \# Noncompliant {{OAR082: The string properties of the specified parameters must define a byte or binary format}}
                      type: string 
                      format: int128
                    line: \# Noncompliant {{OAR082: The string properties of the specified parameters must define a byte or binary format}}
                      type: string 
                    price:  \# Noncompliant {{OAR082: The string properties of the specified parameters must define a byte or binary format}}
                      type: string
                      format: byte

Compliant Solution (OpenAPI 3)
------------------------------

  openapi: "3.0.0"
  info:
    version: "1.0.0"
    title: "Swagger Petstore"
  paths:
    /invoices:
      get:
        responses:
          '200':
            description: A invoice.
            content: 
              application/json:
                schema:
                  type: object
                  properties:
                    producto:  
                    type: string
                    format: byte
                    linea: 
                      type: string
                      format: byte 
                    precio:
                      type: string
                      format: byte