When defining an API, the request data formats must be defined.

For **OpenAPI 2** specify at least a MIME Type understood by the API in the "consumes" keyword is mandatory.

For **OpenAPI 3** specify at least a Media Type in the content of the response body is mandatory.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    put: \# Noncompliant {{OAR007: Section produces is mandatory}}
      responses: 
        200:
          description: OK

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
produces:
  - application/json
paths:
  /pets:
    put:
      responses: 
        200:
          description: OK

Or

swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    put:
      produces:
        - application/json
      responses: 
        200:
          description: OK

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: 3.0.0
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    post: \# Noncompliant {{OAR007: Section produces is mandatory}}
      responses:
        200:
          description: Ok

Compliant Solution (OpenAPI 3)
------------------------------

openapi: 3.0.0
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    post:
      responses:
        200:
          description: Ok
          content:
            application/json:
              schema:
                type: object