A string type with invalid format or without format may cause developers to use the wrong variable types in the API implementation.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /invoices:
    get:
      responses:
        200:
          description: Invoices.
          schema:
            type: array
            items:
              type: object
              properties:
                date:
                  type: string \# Noncompliant {{OAR037: String types requires a valid format}}
                  format: 'dd/mm/yyyy'

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /invoices:
    get:
      responses:
        200:
          description: Invoices.
          schema:
            type: array
            items:
              type: object
              properties:
                date:
                  type: string
                  format: date

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /invoices:
    get:
      responses:
        200:
          description: Invoices.
          content:
            application/json:
              schema:
                type: array
                items:
                  type: object
                  properties:
                    date:
                      type: string \# Noncompliant {{OAR037: String types requires a valid format}}
                      format: dd/mm/yyyy

Compliant Solution (OpenAPI 3)
------------------------------

openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /invoices:
    get:
      responses:
        200:
          description: Invoices.
          content:
            application/json:
              schema:
                type: array
                items:
                  type: object
                  properties:
                    date:
                      type: string
                      format: date