API name must start with the prefix 'api-'

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
basePath: /seguros/v1 \# Noncompliant {{OAR099: API name must start with prefix 'api-'}}
paths: {}

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
    version: 1.0.0
    title: Swagger Petstore
basePath: /api-seguros/v1
paths: {}

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
servers:
  - url: https://petstore.swagger.io/seguros/v1 \# Noncompliant {{OAR099: API name must start with prefix 'api-'}}
paths: {}

Compliant Solution (OpenAPI 3)
------------------------------

openapi: "3.0.0"
info:
    version: 1.0.0
    title: Swagger Petstore
servers:
    - url: https://petstore.swagger.io/api-seguros/v1
paths: {}