The chosen parameter must be defined in this operation. By default, $limit

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /pets:
      get:
        parameters: 
          - in: query
            name: other \# Noncompliant {{OAR025: $limit must be defined as a parameter in this operation}}
            type: array
            items:
              type: string
          - in: query
            name: hola \# Noncompliant {{OAR025: $limit must be defined as a parameter in this operation}}
            type: array
            items:
              type: string
        responses:
          206:
            description: Ok

Compliant Solution (OpenAPI 2)
------------------------------

  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /pets:
      get:
        parameters:
          - in: query
            name: $limit
            type: array
            items:
              type: string
          - in: query
            name: $limit
            type: array
            items:
              type: string
        responses:
          206:
            description: Ok

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

  openapi: "3.0.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /pets:
      get:
        parameters: 
          - in: query
            name: other \# Noncompliant {{OAR025: $limit must be defined as a parameter in this operation}}
            schema:
              type: array
              items:
                type: string
          - in: query
            name: select \# Noncompliant {{OAR025: $limit must be defined as a parameter in this operation}}
            schema:
              type: array
              items:
                type: string
        responses:
          206:
            description: Ok

Compliant Solution (OpenAPI 3)
------------------------------

  openapi: "3.0.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /pets:
      get:
        parameters:
          - in: query
            name: $limit
            schema:
              type: array
              items:
                type: string
          - in: query
            name: $limit
            schema:
              type: array
              items:
                type: string
        responses:
          206:
            description: Ok