Any param in PATH or QUERY, should have bad request (400) response

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

  swagger: "2.0"
  info:
    version: "1.0.0"
    title: "Swagger Petstore"
  paths:
    /items: 
      post: 
        parameters: 
          - name: "param1" # Noncompliant {{OAR069: Any param in PATH or QUERY should have a Bad Request (400) response.}}
            in: "query"
            required: false
            type: "string"
        responses:
          '200':
            description: "A list of items"
            schema:
              type: "array"
              items:
                type: "string"

Compliant Solution (OpenAPI 2)
-------------------------------------

  swagger: '2.0'
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /items:
      get:
        summary: Get a list of items
        parameters:
          - name: param1
            in: query
            required: false
            type: string
        responses:
          200:
            description: A list of items
            schema:
              type: array
              items:
                type: string
          400:
            description: Bad Request

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

  openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /items:
    get:
      summary: Get a list of items
      parameters:
        - name: param1 # Noncompliant {{OAR069: Any param in PATH or QUERY should have a Bad Request (400) response.}}
          in: query
          required: false
          schema:
            type: string
      responses:
        200:
          description: A list of items
          content:
            application/json:
              schema:
                type: array
                items:
                  type: string

Compliant Solution (OpenAPI 3)
-------------------------------------

  openapi: "3.0.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /items:
      get:
        summary: Get a list of items
        parameters:
          - name: param1
            in: query
            required: false
            schema:
              type: string
        responses:
          200:
            description: A list of items
            content:
              application/json:
                schema:
                  type: array
                  items:
                    type: string
          400:
            description: Bad Request