A wrong scope definition may cause problems to import the API definition into WSO2.

Noncompliant Code Example (OpenAPI 2)
---------------------------------------------------------------------------------------

swagger: "2.0"
x-wso2-security:
  apim:
    x-wso2-scopes:
    - name: read
      roles: READ_ROLE
    - name: write
      roles:

Compliant Solution (OpenAPI 2)
--------------------------------------------------------------------------------------

swagger: "2.0"
x-wso2-security:
  apim:
    x-wso2-scopes:
    - name: read
      key: read
      roles: READ_ROLE
    - name: write
      key: write
      roles: WRITE_ROLE

Noncompliant Code Example (OpenAPI 3)
--------------------------------------------------------------------------------------

openapi: "3.0.0"
x-wso2-security:
  apim:
    x-wso2-scopes:
    - name: read
      roles: READ_ROLE
    - name: write
      roles:

Compliant Solution (OpenAPI 3)
--------------------------------------------------------------------------------------

openapi: "3.0.0"
x-wso2-security:
  apim:
    x-wso2-scopes:
    - name: read
      key: read
      roles: READ_ROLE
    - name: write
      key: write
      roles: WRITE_ROLE