The OpenAPI version should be one of the allowed by the organization

### In case 3.0.2 is not allowed

Compliant Solution (OpenAPI 2)
------------------------------

  swagger: "2.0"
  info:
    title: Sample API
    description: This is a sample Swagger API.
    version: "1.0.0"
  paths: {} 

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

  openapi: 3.0.2  \# Noncompliant {{OAR085: The OpenAPI version should be one of the allowed by the organization}}
  info:
    title: Sample API
    description: This is a sample API.
    version: 1.0.0
  paths: {}

Compliant Solution (OpenAPI 3)
------------------------------

  openapi: 3.0.0 
  info:
    title: Sample API
    description: This is a sample API.
    version: 1.0.0
  paths: {}