Response codes in PUT operations must be defined according to the standard

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /status:
      put:
        summary: Get API status
        responses:
          400:
            description: error
    /another:
      put:
        summary: Get another API status
        responses:
          400:
            description: error
    /users:
      put:
        summary: Get all users
        responses:  \# Noncompliant {{OAR063: Response code 200, 202, 204, 206 must be defined}}
          400:
            description: Error      

Compliant Solution (OpenAPI 2)
------------------------------

  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /status:
      put:
        summary: Get API status
        responses:
          200:
            description: OK
          202:
            description: Accepted
          206:
            description: Partial Content
    /users:
      put:
        summary: Get all users
        responses:
          200:
            description: OK
          202:
            description: Accepted
          206:
            description: Partial Content  

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

  openapi: "3.0.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /status:
      put:
        summary: Get API status
        responses:
          400:
            description: error
    /another:
      put:
        summary: Get another API status
        responses:
          400:
            description: error
    /users:
      put:
        summary: Get all users
        responses: \# Noncompliant {{OAR063: Response code 200, 202, 204, 206 must be defined}}
          400:
            description: Error

Compliant Solution (OpenAPI 3)
------------------------------

  openapi: "3.0.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /status:
      put:
        summary: Get API status
        responses:
          200:
            description: OK
          202:
            description: Accepted
          206:
            description: Partial Content
    /users:
      put:
        summary: Get all users
        responses:
          200:
            description: OK
          202:
            description: Accepted
          206:
            description: Partial Content