When defining an API, the response data formats must be defined.

According to the standard, the default format will be application/json, with some exceptions.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
produces: \# Noncompliant {{OAR010: Should indicate the default response media type}}
  - application/xml 
paths:
  /pets:
    put:
      responses:
        200:
          description: Ok

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
produces:
  - application/json
paths:
  /pets:
    put:
      responses:
        200:
          description: Ok

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: 3.0.0
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    post:
      responses:
        200:
          description: Ok
          content: \# Noncompliant {{OAR010: Should indicate the default response media type}}
            application/xml:
              schema:
                type: object

Compliant Solution (OpenAPI 3)
------------------------------

openapi: 3.0.0
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    post:
      responses:
        200:
          description: Ok
          content:
            application/json:
              schema:
                type: object