API should include a 429 'Too Many Requests' response to indicate rate limiting

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

    swagger: '2.0'
    info:
      version: 1.0.0
      title: Non-compliant API
    paths:
      /items:
        get:
          summary: Get a list of items
          responses: \# Noncompliant {{OAR073: API should include a 429 response to indicate rate limiting}}
            '200':
              description: A list of items
              schema:
                type: array
                items:
                  type: string

Compliant Solution (OpenAPI 2)
------------------------------

    swagger: '2.0'
    info:
      version: 1.0.0
      title: Compliant API
    paths:
      /items:
        get:
          summary: Get a list of items
          responses:
            '200':
              description: A list of items
              schema:
                type: array
                items:
                  type: string
            '429':
              description: Too Many Requests
              schema:
                $ref: '#/definitions/error\_response'

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

    openapi: "3.0.0"
    info:
      version: 1.0.0
      title: Non-compliant API
    paths:
      /items:
        get:
          summary: Get a list of items
          responses: \# Noncompliant {{OAR073: API should include a 429 response to indicate rate limiting}}
            200:
              description: A list of items
              content:
                application/json:
                  schema:
                    type: array
                    items:
                      type: string

Compliant Solution (OpenAPI 3)
------------------------------

    openapi: "3.0.0"
    info:
      version: 1.0.0
      title: Compliant API
    paths:
      /items:
        get:
          summary: Get a list of items
          responses:
            200:
              description: A list of items
              content:
                application/json:
                  schema:
                    type: array
                    items:
                      type: string
            429:
              description: Too Many Requests
              content:
                application/json:
                  schema:
                    $ref: '#/components/schemas/error\_response'