String parameters should have minimum length, maximum length, regular expression, or enum restriction.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

  swagger: '2.0'
  info:
    title: API sin restricciones
    version: 1.0.0
  paths:
    /users/{id}:
      get:
        parameters:
          - name: id
            in: path
            required: true
            type: string \# Noncompliant {{OAR075: String parameters should have minLength, maxLength, pattern (regular expression), or enum restriction}}
        responses:
          '200':
            description: Un usuario

Compliant Solution (OpenAPI 2)
------------------------------

  swagger: '2.0'
  info:
    version: 1.0.0
    title: My API
  paths:
    /items:
      get:
        summary: Get a list of items
        parameters:
          - name: param1
            in: query
            type: string
            minLength: 1
            maxLength: 10
            enum: \['option1', 'option2'\]
        responses:
          200:
            description: A list of items
            schema:
              type: array
              items:
                type: string

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

  openapi: 3.0.0
  info:
    title: API sin restricciones
    version: 1.0.0
  paths:
    /users/{id}:
      get:
        parameters:
          - name: id
            in: path
            required: true
            schema:
              type: string \# Noncompliant {{OAR075: String parameters should have minLength, maxLength, pattern (regular expression), or enum restriction}}
        responses:
          '200':
            description: Un usuario

Compliant Solution (OpenAPI 3)
------------------------------

  openapi: "3.0.0"
  info:
    version: 1.0.0
    title: My API
  paths:
    /items:
      get:
        summary: Get a list of items
        parameters:
          - name: param1
            in: query
            schema:
              type: string
              minLength: 1
              maxLength: 10
              enum: \['option1', 'option2'\]
        responses:
          200:
            description: A list of items
            content:
              application/json:
                schema:
                  type: array
                  items:
                    type: string