Base path must have at least two parts

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
basePath: /api-seguros \# Noncompliant {{OAR097: Path has to few parts}}
paths: {}

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
    version: 1.0.0
    title: Swagger Petstore
basePath: /api-seguros/v1
paths: {}

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: "3.0.0"
info:
    version: 1.0.0
    title: Swagger Petstore
servers:
    - url: https://petstore.swagger.io/api-seguros \# Noncompliant {{OAR097: Path has to few parts}}
paths: {}

Compliant Solution (OpenAPI 3)
------------------------------

openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
servers:
  - url: https://petstore.swagger.io/api-seguros/v1
paths: {}