The use of session mechanisms with state is forbidden. This may cause problems to scale the applciations.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /pets:
    get:
      parameters:
        - in: header
          name: Cookie
          type: string
          required: true
      responses:
        200:
          description: Ok
          headers:
            Set-Cookie:
              type: string
              description: session

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /pets:
    get:
      parameters:
        - in: header
          name: X-Authorization-Key
          type: string
          required: true
      responses:
        200:
          description: Ok

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /pets:
    get:
      parameters:
      - name: Cookie
        in: header
        required: true
        schema:
          type: string
      responses:
        200:
          description: Ok
          headers:
            Set-Cookie:
              description: session
              schema:
                type: string

Compliant Solution (OpenAPI 3)
------------------------------

openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /pets:
    get:
      parameters:
      - name: X-Authorization-Key
        in: header
        required: true
        schema:
          type: string
      responses:
        200:
          description: Ok