ALL Api methods should have security

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

  swagger: "2.0"
  info:
    version: "1.0.0"
    title: "Sample API"
  securityDefinitions:
    apiKey:
      type: apiKey
      name: X-API-Key
      in: header
  paths:
    /users:
      get: \# Noncompliant {{OAR078: All API methods must have security}}
        responses:
          200:
            description: "OK"

Compliant Solution (OpenAPI 2)
------------------------------

  swagger: "2.0"
  info:
    version: "1.0.0"
    title: Sample API
  paths:
    /users:
      get:
        security:
          - apiKey: \[\]
        responses:
          "200":
            description: OK

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

  openapi: 3.0.0
  info:
    version: 1.0.0
    title: Sample API
  components:
    securitySchemes:
      apiKey:
        type: apiKey
        name: X-API-Key
        in: header
  paths:
    /users:
      get: \# Noncompliant {{OAR078: All API methods must have security}}
        responses:
          '200':
            description: OK

Compliant Solution (OpenAPI 3)
------------------------------

  openapi: 3.0.0
  info:
    version: 1.0.0
    title: Sample API
  paths:
    /users:
      get:
        security:
          - apiKey: \[\]
        responses:
          '200':
            description: OK