The security scheme must be among those allowed by the organization and must be complete.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

    swagger: "2.0"
    info:
    version: "1.0.0"
    title: "Sample API"
    securityDefinitions:
    apiKey:
        type: apiKey
        name: X-API-Key
        in: header
    paths:
    /users:
        get: \# Noncompliant {{OAR080: The security scheme must be among those allowed by the organization and must be complete}}
        responses:
            200:
            description: "OK"
    /products:
        post: \# Noncompliant {{OAR080: The security scheme must be among those allowed by the organization and must be complete}}
        responses:
            200:
            description: "OK"
    /orders:
        put: \# Noncompliant {{OAR080: The security scheme must be among those allowed by the organization and must be complete}}
        responses:
            200:
            description: "OK"
    /invoices:
        delete: \# Noncompliant {{OAR080: The security scheme must be among those allowed by the organization and must be complete}}
        responses:
            200:
            description: "OK"

Compliant Solution (OpenAPI 2)
------------------------------

    swagger: "2.0"
    info:
      version: 1.0.0
      title: Sample API
    paths:
      /users:
        get:
          summary: Get all users
          security:
            - apiKey: \[\]
            - oauth2: \[\]
          responses:
            '200':
              description: OK
      /products:
        post:
          summary: Create a new product
          security:
            - oauth2: \[\]
            - apiKey: \[\]
          responses:
            '200':
              description: OK
    securityDefinitions:
      oauth2:
        type: oauth2
        flow: implicit
        authorizationUrl: https://example.com/oauth/authorize
        scopes:
          read: Read access
          write: Write access
      apiKey:
        type: apiKey
        name: X-API-Key
        in: header

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

    openapi: 3.0.0
    info:
      version: 1.0.0
      title: Sample API
    paths:
      /users:
        get: \# Noncompliant {{OAR080: The security scheme must be among those allowed by the organization and must be complete}}
          summary: Get all users
          security: 
            - apiKey: \[\]
          responses:
            '200':
              description: OK
      /products:
        post: \# Noncompliant {{OAR080: The security scheme must be among those allowed by the organization and must be complete}}
          summary: Create a new product
          security: 
            - oauth2: \[\]
          responses:
            '200':
              description: OK
      /orders:
        put: \# Noncompliant {{OAR080: The security scheme must be among those allowed by the organization and must be complete}}
          summary: Update an order
          security: 
            - unknownScheme: \[\]
          responses:
            '200':
              description: OK
      /invoices:
        delete: \# Noncompliant {{OAR080: The security scheme must be among those allowed by the organization and must be complete}}
          summary: Delete an invoice
          security: 
            - unknownScheme: \[\]
          responses:
            '200':
              description: OK

Compliant Solution (OpenAPI 3)
------------------------------

    openapi: 3.0.0
    info:
      version: 1.0.0
      title: Sample API
    paths:
      /users:
        get:
          summary: Get all users
          security:
            - apiKey: \[\]
            - oauth2: \[\]
          responses:
            '200':
              description: OK
      /products:
        post:
          summary: Create a new product
          security:
            - apiKey: \[\]
            - oauth2: \[\]
          responses:
            '200':
              description: OK
      /orders:
        put:  
          summary: Update an order
          security: 
            - unknownScheme: \[\] 
          responses:
            '200':
              description: OK
      /invoices:
        delete: 
          summary: Delete an invoice
          security: 
            - unknownScheme: \[\] 
          responses:
            '200':
              description: OK