The $ref of a parameter must end with a corresponding suffix

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

  swagger: '2.0'
info:
  title: API de ejemplo
  version: '1.0'
paths:
  /mascotas:
    get:
      parameters:
        - $ref: '#/parameters/idMascotaRef' \# Noncompliant {{OAR088: The $ref of a parameter must end with the suffix Param}}
      responses:
        '200':
          description: Lista de mascotas
parameters:
  idMascotaRef:
    name: idMascota
    in: query
    required: true
    type: string

Compliant Solution (OpenAPI 2)
------------------------------

  swagger: '2.0'
  info:
    title: API de ejemplo
    version: '1.0'
  paths:
    /mascotas:
      get:
        parameters:
          - $ref: '#/parameters/idMascotaParam'
        responses:
          '200':
            description: Lista de mascotas
  parameters:
    idMascotaparam:
      name: idMascota
      in: query
      required: true
      type: string

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

  openapi: 3.0.0
  info:
    title: API de ejemplo
    version: "1.0"
  paths:
    /mascotas:
      get:
        parameters:
          - $ref: '#/components/parameters/idMascotaRef' \# Noncompliant {{OAR088: The $ref of a parameter must end with the suffix Param}}
        responses:
          '200':
            description: Lista de mascotas
  components:
    parameters:
      idMascotaRef:
        name: idMascota
        in: query
        required: true
        schema:
          type: string

Compliant Solution (OpenAPI 3)
------------------------------

  openapi: 3.0.0
  info:
    title: API de ejemplo
    version: "1.0"
  paths:
    /mascotas:
      get:
        parameters:
          - $ref: '#/components/parameters/idMascotaParam'
        responses:
          '200':
            description: Lista de mascotas
  components:
    parameters:
      idMascotaParam:
        name: idMascota
        in: query
        required: true
        schema:
          type: string