  A scope name non compliant with the standard may cause problems at application level.

  Noncompliant Code Example (OpenAPI 2)
  ---------------------------------------------------------------------------------------

  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  x-wso2-security:
    apim:
      x-wso2-scopes:
      - name: read
        roles: READ_ROLE
        key: read
        description: Allows users to read data

  Compliant Solution (OpenAPI 2)
  ---------------------------------------------------------------------------------------

  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  x-wso2-security:
    apim:
      x-wso2-scopes:
      - name: apfc_sc_read
        roles: READ_ROLE
        key: read
        description: Allows users to read data

  Noncompliant Code Example (OpenAPI 3)
  ---------------------------------------------------------------------------------------

  openapi: "3.0.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  x-wso2-security:
    apim:
      x-wso2-scopes:
      - name: read
        roles: READ_ROLE
        key: read
        description: Allows users to read data

  Compliant Solution (OpenAPI 3)
  ---------------------------------------------------------------------------------------

  openapi: "3.0.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  x-wso2-security:
    apim:
      x-wso2-scopes:
      - name: apfc_sc_read
        roles: READ_ROLE
        key: read
        description: Allows users to read data