It is validated that each route alternates its parts between static and variable parts, starting with the static parts.

Examples:

*   /{id} -> Error
*   /facturas/{id}/{lineId} -> Error
*   /facturas/{id} -> Ok
*   /facturas/get -> Ok
*   /facturas/{id}/lineas/{lineId} -> Ok
*   /facturas/{id}/lineas/get -> Ok

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

```yaml
swagger: "2.0"
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /customers/{customerId}/{invoiceId}:
    get:
      parameters:
        - in: path
          name: customerId
          type: integer
          required: true
        - in: path
          name: invoiceId
          type: integer
          required: true
      responses:
        200:
          description: Ok

```

Compliant Solution (OpenAPI 2)
------------------------------

```yaml
swagger: "2.0"
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /customers/{customerId}/invoices:
    get:
      parameters:
        - in: path
          name: customerId
          type: integer
          required: true
      responses:
        200:
          description: Ok
  /invoices/{invoiceId}:
    get:
      parameters:
        - in: path
          name: invoiceId
          type: integer
          required: true
      responses:
        200:
          description: Ok

```

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

```yaml
openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /customers/{customerId}/{invoiceId}:
    get:
      parameters:
      - name: customerId
        in: path
        required: true
        schema:
          type: integer
      - name: invoiceId
        in: path
        required: true
        schema:
          type: integer
      responses:
        200:
          description: Ok

```

Compliant Solution (OpenAPI 3)
------------------------------

```yaml
openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /customers/{customerId}/invoices:
    get:
      parameters:
      - name: customerId
        in: path
        required: true
        schema:
          type: integer
      responses:
        200:
          description: Ok
  /invoices/{invoiceId}:
    get:
      parameters:
      - name: invoiceId
        in: path
        required: true
        schema:
          type: integer
      responses:
        200:
          description: Ok

```