The defined media types must conform to the [RFC6838](https://tools.ietf.org/html/rfc6838) and [RFC7231](https://tools.ietf.org/html/rfc7231). This ensures that a tool can rely on the declared media type to generate appropriate code to handle the parsing of output.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
    version: 1.0.0
    title: Swagger Petstore
produces:
    - application/json
    - text/\* \# Noncompliant {{OAR044: Declared mime type should conform to RFC6838}}
consumes:
    - application/json
    - text/\* \# Noncompliant {{OAR044: Declared mime type should conform to RFC6838}}
paths:
    /pets:
    get:
        produces:
        - application
        - text/plain \# Noncompliant {{OAR044: Declared mime type should conform to RFC6838}}
        consumes:
        - application
        - text/plain \# Noncompliant {{OAR044: Declared mime type should conform to RFC6838}}
        responses:
        '200':
            description: some operation

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
    version: 1.0.0
    title: Swagger Petstore
produces:
    - application/json
    - text/plain
consumes:
    - application/json
    - text/plain
paths:
    /pets:
    get:
        produces:
        - application/json
        - text/plain
        consumes:
        - application/json
        - text/plain
        responses:
        '200':
            description: some operation

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: "3.0.1"
info:
    version: 1.0.0
    title: Swagger Petstore
paths:
    /pets:
    post:
        requestBody:
        content:
            'application': {} \# Noncompliant {{OAR044: Declared media type range should conform to RFC7231}}
            'text/\*': {} 
        responses:
        '200':
            description: some operation
            content:
            'application': {} \# Noncompliant {{OAR044: Declared media type range should conform to RFC7231}}

Compliant Solution (OpenAPI 3)
------------------------------

openapi: "3.0.1"
info:
    version: 1.0.0
    title: Swagger Petstore
paths:
    /pets:
    post:
        requestBody:
        content:
            'application/json': {}
            'text/plain': {}
        responses:
        '200':
            description: some operation
            content:
            'application/json': {}