A description can help other developers to understand the correct use of the scope.

Noncompliant Code Example (OpenAPI 2)
---------------------------------------------------------------------------------------

swagger: "2.0"
x-wso2-security:
  apim:
    x-wso2-scopes:
    - name: read
      roles: READ_ROLE
      key: read

Compliant Solution (OpenAPI 2)
---------------------------------------------------------------------------------------

swagger: "2.0"
x-wso2-security:
  apim:
    x-wso2-scopes:
    - name: read
      roles: READ_ROLE
      key: read
      description: Allows users to read data

Noncompliant Code Example (OpenAPI 3)
---------------------------------------------------------------------------------------

openapi: "3.0.0"
x-wso2-security:
  apim:
    x-wso2-scopes:
    - name: read
      roles: READ_ROLE
      key: read

Compliant Solution (OpenAPI 3)
---------------------------------------------------------------------------------------

openapi: "3.0.0"
x-wso2-security:
  apim:
    x-wso2-scopes:
    - name: read
      roles: READ_ROLE
      key: read
      description: Allows users to read data