The default value of $total parameter should be false to prevent performance issues.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /pets:
      get:
        parameters:
          - in: query
            name: other
            type: array
            items:
              type: string
          - in: query
            name: $total
            type: boolean
            default: true \# Noncompliant {{OAR026: The $total parameter default value should be false}}
        responses:
          206:
            description: Ok

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /pets:
    get:
      parameters:
        - in: query
          name: $total
          required: false
          type: boolean
          default: false
      responses:
        200:
          description: Ok

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

  openapi: "3.0.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /pets:
      get:
        parameters:
          - in: query
            name: other
            schema:
              type: array
              items:
                type: string
          - in: query
            name: $total
            schema:
              type: boolean
              default: true \# Noncompliant {{OAR026: The $total parameter default value should be false}}
        responses:
          206:
            description: Ok

Compliant Solution (OpenAPI 3)
------------------------------

openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /pets:
    get:
      parameters:
        - in: query
          name: $total
          required: false
          schema:
            type: boolean
            default: false
      responses:
        200:
          description: Ok