POST creation responses must have a Location header with the reference to the new resource.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /pets:
    post:
      responses:
        201: \# Noncompliant {{OAR027: Location header is required in responses with code 201 from POST operations}}
          description: Created

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /pets:
    post:
      responses:
        201:
          description: Created
          headers:
            Location:
              type: string

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: 3.0.0
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /pets:
    post:
      responses:
        201: \# Noncompliant {{OAR027: Location header is required in responses with code 201 from POST operations}}
          description: Created

Compliant Solution (OpenAPI 3)
------------------------------

openapi: 3.0.0
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /pets:
    post:
      responses:
        201:
          description: Created
          headers:
            Location:
              schema:
                type: string