For resources that must be named with two or more words it is recommended to use the naming convention "kebab-case" (lowercase words separated by a hyphen) to increase the readability of the url.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /Invoices/{id}/CUSTOM\_DETAIL: \# Noncompliant {{OAR011: The base path and resource names with more than two words must be compliant with the standard naming convention: kebab-case}}
    get:
      parameters:
        - in: path
          name: id
          type: integer
          required: true
      responses:
        200:
          description: Ok

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /invoices/{id}/custom-detail:
    get:
      parameters:
        - in: path
          name: id
          type: integer
          required: true
      responses:
        200:
          description: Ok

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: "3.0.0"
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /Invoices/{id}/CUSTOM\_DETAIL: \# Noncompliant {{OAR011: The base path and resource names with more than two words must be compliant with the standard naming convention: kebab-case}}
    get:
      parameters:
        - in: path
          name: id
          schema:
            type: integer
          required: true
      responses:
        200:
          description: Ok

Compliant Solution (OpenAPI 3)
------------------------------

openapi: "3.0.0"
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /invoices/{id}/custom-detail:
    get:
      parameters:
        - in: path
          name: id
          schema:
            type: integer
          required: true
      responses:
        200:
          description: Ok