An API without a status endpoint is hard to observe with monitoring tools.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths: \# Noncompliant {{OAR030: The path '/status' must be declared}}
  /pets:
    get:
      responses:
        200:
          description: Ok

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /pets:
    get:
      responses:
        200:
          description: Ok
  /status:
    get:
      responses:
        200:
          description: Ok

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: 3.0.0
info:
  version: 1.0.0
  title: Swagger Petstore
paths: \# Noncompliant {{OAR030: The path '/status' must be declared}}
  /pets:
    get:
      responses:
        200:
          description: Ok

Compliant Solution (OpenAPI 3)
------------------------------

openapi: 3.0.0
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /pets:
    get:
      responses:
        200:
          description: Ok
  /status:
    get:
      responses:
        200:
          description: Ok