When defining an API, the request data formats must be defined.

For **OpenAPI 2** specify at least a MIME Type understood by the API in the "consumes" keyword is mandatory.

For **OpenAPI 3** specify at least a Media Type in the content of the request body is mandatory.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    put: \# Noncompliant {{OAR006: Section consumes is mandatory}}
      responses: 
        200:
          description: OK

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
consumes:
  - application/json
paths:
  /pets:
    put:
      responses: 
        200:
          description: OK

Or

swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    put:
      consumes:
        - application/json
      responses: 
        200:
          description: OK

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: 3.0.0
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    post: \# Noncompliant {{OAR006: Section consumes is mandatory}}
      responses:
        200:
          description: Ok

Compliant Solution (OpenAPI 3)
------------------------------

openapi: 3.0.0
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    post:
      requestBody:
        description: Optional description in \*Markdown\*
        required: true
        content:
          application/json:
            schema:
              type: object
      responses:
        200:
          description: Ok