Some formats should not pass through some querystring

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

### In case format: password is forbidden

    swagger: '2.0'
    info:
      version: 1.0.0
      title: My API
    paths:
      /examples:
        get:
          summary: Get a list of items
          parameters:
            - name: exampleParam1
              in: query
              type: string
              format: password  \# Noncompliant {{OAR084: The format password should not pass through this querystring}}
            - name: param3
              in: query
              type: boolean
          responses:
            200:
              description: A list of items
              schema:
                type: array
                items:
                  type: string    

Compliant Solution (OpenAPI 2)
------------------------------

    swagger: '2.0'
    info:
      version: 1.0.0
      title: My API
    paths:
      /examples:
        get:
          summary: Get a list of items
          parameters:
            - name: exampleParam1
              in: query
              type: string
              format: date 
            - name: param3
              in: query
              type: boolean
          responses:
            200:
              description: A list of items
              schema:
                type: array
                items:
                  type: string    

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

    openapi: 3.0.0
    info:
      version: 1.0.0
      title: My API
    paths:
      /examples:
        get:
          summary: Get a list of items
          parameters:
            - name: exampleParam1
              in: query
              schema:
                type: string
                format: password  \# Noncompliant {{OAR084: The format password should not pass through this querystring}}
            - name: param3
              in: query
              schema:
                type: boolean
          responses:
            200:
              description: A list of items
              content:
                application/json:
                  schema:
                    type: array
                    items:
                      type: string

Compliant Solution (OpenAPI 3)
------------------------------

    openapi: 3.0.0
    info:
      version: 1.0.0
      title: My API
    paths:
      /examples:
        get:
          summary: Get a list of items
          parameters:
            - name: exampleParam1
              in: query
              schema:
                type: string
                format: date  
            - name: param3
              in: query
              schema:
                type: boolean
          responses:
            200:
              description: A list of items
              content:
                application/json:
                  schema:
                    type: array
                    items:
                      type: string