Define the "default" response is mandatory.

The "default" response is used for all HTTP codes that are not covered individually.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    get:
      responses: \# Noncompliant {{OAR013: Default response is required}}
        200:
          description: OK

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    get:
      responses:
        200:
          description: OK
        default:
          description: Error

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: 3.0.1
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    get:
      responses: \# Noncompliant {{OAR013: Default response is required}}
        200:
          description: OK

Compliant Solution (OpenAPI 3)
------------------------------

openapi: 3.0.1
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    get:
      responses:
        200:
          description: OK
        default:
          description: Error