Numeric parameters should have minimum, maximum, or format restriction

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

  swagger: '2.0'
  info:
    version: 1.0.0
    title: My API
  paths:
    /items:
      get:
        summary: Get a list of items
        parameters:
          - name: param1
            in: query
            type: integer \# Noncompliant {{OAR074: Numeric parameters should have minimum, maximum, or format restriction}}
          - name: param2
            in: query
            type: number  \# Noncompliant {{OAR074: Numeric parameters should have minimum, maximum, or format restriction}}
            minimum: 12
        responses:
          200:
            description: A list of items

Compliant Solution (OpenAPI 2)
------------------------------

  swagger: '2.0'
  info:
    version: 1.0.0
    title: My API
  paths:
    /items:
      get:
        summary: Get a list of items
        parameters:
          - name: param1
            in: query
            type: integer
            minimum: 0
            maximum: 100
        responses:
          200:
            description: A list of items
            schema:
              type: array
              items:
                type: string

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

  openapi: "3.0.0"
info:
  version: 1.0.0
  title: My API
paths:
  /items:
    get:
      summary: Get a list of items
      parameters:
        - name: param1
          in: query
          schema:
            type: integer \# Noncompliant {{OAR074: Numeric parameters should have minimum, maximum, or format restriction}}
        - name: param2
          in: query
          schema:
            type: number \# Noncompliant {{OAR074: Numeric parameters should have minimum, maximum, or format restriction}}
      responses:
        200:
          description: A list of items

Compliant Solution (OpenAPI 3)
------------------------------

  openapi: "3.0.0"
  info:
    version: 1.0.0
    title: My API
  paths:
    /items:
      get:
        summary: Get a list of items
        parameters:
          - name: param1
            in: query
            schema:
              type: integer
              minimum: 0
              maximum: 100
        responses:
          200:
            description: A list of items
            content:
              application/json:
                schema:
                  type: array
                  items:
                    type: string