Define a summary for each operation. The OpenAPI contract is a way to convey meaning to non-technical users. The summary will help those populations understand what the operation is about, instead of approaching it in technical terms.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /pets:
    post: \# Noncompliant {{OAR050: Provide a summary for each operation}}
      responses:
        default:
          description: the default response
    get: \# Noncompliant {{OAR050: Provide a summary for each operation}}
      responses:
        default:
          description: the default response

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /pets:
    post:
      summary: Create a new pet
      responses:
        default:
          description: the default response
    get:
      summary: List all pets
      responses:
        default:
          description: the default response

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: 3.0.0
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /pets:
    post: \# Noncompliant {{OAR050: Provide a summary for each operation}}
      responses:
        default:
          description: the default response
    get: \# Noncompliant {{OAR050: Provide a summary for each operation}}
      responses:
        default:
          description: the default response

Compliant Solution (OpenAPI 3)
------------------------------

openapi: 3.0.0
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /pets:
    post:
      summary: Create a new pet
      responses:
        default:
          description: the default response
    get:
      summary: List all pets
      responses:
        default:
          description: the default response