Base path must be compliant with the standard. 
The base path must be composed of exactly two parts:
- The first part corresponds to the API name and must start with the prefix api-, followed by at least one character.
- The second part corresponds to the API version and must start with the prefix v, followed by an integer number.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

``` yaml
swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
basePath: /seguros
paths: {}
```

``` yaml
swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
basePath: /api-seguros/v1/vida
paths: {}
```

``` yaml
swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
basePath: /api-seguros/v1.0
paths: {}
```

Compliant Solution (OpenAPI 2)
------------------------------

``` yaml
swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
basePath: /api-seguros/v1
paths: {}
```

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

``` yaml
openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
servers:
  - url: https://petstore.swagger.io/seguros/version/1
paths: {}
```

``` yaml
openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
servers:
  - url: https://petstore.swagger.io/api-seguros/v1/vida
paths: {}
```

``` yaml
openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
servers:
  - url: https://petstore.swagger.io/api-seguros/v1.0
paths: {}
```

Compliant Solution (OpenAPI 3)
------------------------------

``` yaml
openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
servers:
  - url: https://petstore.swagger.io/api-seguros/v1
paths: {}
```