Paths parameters should have not found (404) response

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

  swagger: '2.0'
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /items:
    get:
      summary: Get a list of items
      parameters:
        - name: param1
          in: path
          required: false
          type: string
      responses: \# Noncompliant {{OAR079: Paths parameters should have not found (404) response}}
        200:
          description: A list of items
          schema:
            type: array
            items:
              type: string

Compliant Solution (OpenAPI 2)
------------------------------

  swagger: '2.0'
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /items:
      get:
        summary: Get a list of items
        parameters:
          - name: param1
            in: path
            required: false
            type: string
        responses: 
          200:
            description: A list of items
            schema:
              type: array
              items:
                type: string
          404:
            description: Not Found

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

  openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /items:
    get:
      summary: Get a list of items
      parameters:
        - name: param1
          in: path
          required: false
          schema:
            type: string
      responses: \# Noncompliant {{OAR079: Paths parameters, should have not found (404) response}}
        200:
          description: A list of items
          content:
            application/json:
              schema:
                type: array
                items:
                  type: string

Compliant Solution (OpenAPI 3)
------------------------------

  openapi: "3.0.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /items:
      get:
        summary: Get a list of items
        parameters:
          - name: param1
            in: path
            required: false
            schema:
              type: string
        responses:
          200:
            description: A list of items
            content:
              application/json:
                schema:
                  type: array
                  items:
                    type: string
          404:
            description: Not Found