Each tag should be documented with a short statement explaining its intent.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
tags:
- name: unused-tag \# Noncompliant {{OAR047: Add a short description to this tag}}
- name: some-tag
paths:
  /pets:
    get:
      tags:
      - some-tag
      responses:
        200:
          description: OK

Compliant Solution (OpenAPI 2)
------------------------------

swaggeer: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
tags:
- name: some-tag
  description: a tag referenced in the operations
- name: other-tag
  description: another tag referenced in the operations
paths:
  /pets:
    get:
      tags:
      - some-tag
      - other-tag
      responses:
        200:
          description: OK

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: "3.0.1"
info:
  version: 1.0.0
  title: Swagger Petstore
tags:
- name: unused-tag \# Noncompliant {{OAR047: Add a short description to this tag}}
- name: some-tag
paths:
  /pets:
    get:
      tags:
      - some-tag
      responses:
        200:
          description: OK

Compliant Solution (OpenAPI 3)
------------------------------

openapi: "3.0.1"
info:
  version: 1.0.0
  title: Swagger Petstore
tags:
- name: some-tag
  description: a tag referenced in the operations
- name: other-tag
  description: another tag referenced in the operations
paths:
  /pets:
    get:
      tags:
      - some-tag
      - other-tag
      responses:
        200:
          description: OK