When defining an API, the request data formats must be defined.

According to the standard, the default format will be application/json, with some exceptions.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
consumes: \# Noncompliant {{OAR009: Should indicate the default request media type}}
  - application/xml
paths:
  /pets:
    put:
      responses: 
        200:
          description: OK

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
consumes:
  - application/json
paths:
  /pets:
    put:
      responses: 
        200:
          description: OK

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: 3.0.0
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    post:
      requestBody:
        description: Optional description in \*Markdown\*
        required: true
        content: \# Noncompliant {{OAR009: Should indicate the default request media type}}
          application/xml:
            schema:
              type: object
      responses:
        200:
          description: Ok

Compliant Solution (OpenAPI 3)
------------------------------

openapi: 3.0.0
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    post:
      requestBody:
        description: Optional description in \*Markdown\*
        required: true
        content:
          application/json:
            schema:
              type: object
      responses:
        200:
          description: Ok