Hostname must be a subdomain of the organization's domain name.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
    version: 1.0.0
    title: Swagger Petstore
host: other.domain.com
paths: {}

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
    version: 1.0.0
    title: Swagger Petstore
host: api.apiaddicts.org
paths: {}

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: "3.0.0"
info:
    version: 1.0.0
    title: Swagger Petstore
servers:
    - url: https://aws-dev-apiaddicts.org/api-users/v1/
    - url: https://aws-test.apiaddicts.org/api-users/v1/
    - url: https://aws-apiaddicts.org/api-users/v1/
paths: {}

Compliant Solution (OpenAPI 3)
------------------------------

openapi: "3.0.0"
info:
    version: 1.0.0
    title: Swagger Petstore
servers:
    - url: https://api-dev.apiaddicts.org/api-users/v1/
    - url: https://api-test.apiaddicts.org/api-users/v1/
    - url: https://api.apiaddicts.org/api-users/v1/
paths: {}