All summaries must begin with a capital letter and end with a period

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

  swagger: '2.0'
  info:
    title: Sample API
    version: 1.0.0
    description: Esta es la descripción de la API.
  paths:
    /pets:
      get:
        summary: Lista de mascotas \# Noncompliant {{OAR087: All summaries must begin with a capital letter and end with a period}}
        description: Esta ruta devuelve una lista de mascotas.
        responses:
          '200':
            description: Una lista de mascotas retornada con éxito.
            schema:
              $ref: '#/definitions/Pet'
  definitions:
    Pet:
      type: object
      properties:
        id:
          type: integer
          format: int64
        name:
          type: string
      description: Representa una mascota.

Compliant Solution (OpenAPI 2)
------------------------------

  swagger: '2.0'
  info:
    title: Sample API
    version: 1.0.0
    description: Esta es la descripción de la API.
  paths:
    /pets:
      get:
        summary: Lista de mascotas.
        description: Esta ruta devuelve una lista de mascotas.
        responses:
          '200':
            description: Una lista de mascotas retornada con éxito.
            schema:
              $ref: '#/definitions/Pet'
  definitions:
    Pet:
      type: object
      properties:
        id:
          type: integer
          format: int64
        name:
          type: string
      description: Representa una mascota.

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

  openapi: 3.0.0
  info:
    title: Sample API
    version: 1.0.0
    description: Esta es la descripción de la API.
    summary: Este es un resumen del API \# Noncompliant {{OAR087: All summaries must begin with a capital letter and end with a period}}
  
  paths:
    /pets:
      get:
        summary: lista de mascotas. \# Noncompliant {{OAR087: All summaries must begin with a capital letter and end with a period}}
        description: Esta ruta devuelve una lista de mascotas.
        responses:
          '200':
            description: Una lista de mascotas retornada con éxito.
  
  components:
    schemas:
      Pet:
        type: object
        properties:
          id:
            type: integer
            format: int64
          name:
            type: string
        description: Representa una mascota.
        summary: Resumen de una mascota \# Noncompliant {{OAR087: All summaries must begin with a capital letter and end with a period}}

Compliant Solution (OpenAPI 3)
------------------------------

  openapi: 3.0.0
  info:
    title: Sample API
    version: 1.0.0
    description: Esta es la descripción de la API.
    summary: Este es un resumen del API.
  
  paths:
    /pets:
      get:
        summary: Lista de mascotas.
        description: Esta ruta devuelve una lista de mascotas.
        responses:
          '200':
            description: Una lista de mascotas retornada con éxito.
  
  components:
    schemas:
      Pet:
        type: object
        properties:
          id:
            type: integer
            format: int64
          name:
            type: string
        description: Representa una mascota.
        summary: Resumen de una mascota.