A resource creation response must have at least one attribute.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /endpoint:
    post:
      responses:
        201:
          description: Ok
          schema: \# Noncompliant {{OAR038: 'data' property is missing}}
            type: object
        204:
          description: No content

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /endpoint:
    post:
      responses:
        201:
          description: Ok
          schema:
            $ref: '#/definitions/response'
        204:
          description: No content

definitions:
  response:
    type: object
    properties:
      data:
        type: object
        properties:
          id:
            type: string

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /endpoint:
    post:
      responses:
        201:
          description: Ok
          content:
            application/json:
              schema: \# Noncompliant {{OAR038: 'data' property is missing}}
                type: object
        204:
          description: No content

Compliant Solution (OpenAPI 3)
------------------------------

openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /endpoint:
    post:
      responses:
        201:
          description: Ok
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/response'
        204:
          description: No content
components:
  schemas:
    response:
      type: object
      properties:
        data:
          type: object
          properties:
            id:
              type: string