A response not compliant with the standard may cause application issues.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /endpoint:
      get:
        responses:
          200:
            description: Ok
            schema:
              $ref: '#/definitions/successResponse'
          204:
            description: No content
          400:
            description: Error
            schema:
              $ref: '#/definitions/errorResponse'
          default:
            description: Unknown
  
  definitions:
    successResponse:
      type: object
      properties:
        payload:
          type: object
          properties:
            tipos:
              type: array
              items:
                type: string
    errorResponse:
      type: object
      properties:
        error:
          type: object
          properties:
            code:
              type: integer \# Noncompliant {{OAR029: 'code' must be of type string}}
            message:
              type: integer \# Noncompliant {{OAR029: 'message' must be of type string}}
            details:
              type: integer \# Noncompliant {{OAR029: 'details' must be of type array}}
            httpStatus:
              type: string \# Noncompliant {{OAR029: 'httpStatus' must be of type integer}}
          required:
            - code
            - message
            - httpStatus

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /endpoint:
    get:
      responses:
        200:
          description: Ok
          schema:
            $ref: '#/definitions/response'
        204:
          description: No content
        400:
          description: Error
          schema:
            $ref: '#/definitions/response'
        default:
          description: Unknown
definitions:
  response:
    type: object
    allOf:
      - $ref: '#/definitions/standard\_response'
    properties:
      data:
        type: object
        properties:
          tipos:
            type: array
            items:
              type: string
    required:
      - data
      - status
  standard\_response:
    type: object
    properties:
      status:
        type: object
        properties:
          http\_status:
            type: string
          code:
            type: integer
          errors:
            type: array
            items:
              type: object
              properties: 
                name:
                  type: string
                value:
                  type: string
          description:
            type: string
          internal\_code:
            type: string
        required:
          - http\_status
          - code
          - description  

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

  openapi: "3.0.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /endpoint:
      get:
        responses:
          200:
            description: Ok
            content:
              application/json:
                schema:
                  $ref: '#/components/schemas/successResponse'
          204:
            description: No content
          400:
            description: Error
            content:
              application/json:
                schema:
                  $ref: '#/components/schemas/errorResponse'
          default:
            description: Unknown
  
  components:
    schemas:
      successResponse:
        type: object
        properties:
          payload:
            type: object
            properties:
              tipos:
                type: array
                items:
                  type: string
      errorResponse:
        type: object
        properties:
          error:
            type: object
            properties:
              code:
                type: integer \# Noncompliant {{OAR029: 'code' must be of type string}}
              message:
                type: integer \# Noncompliant {{OAR029: 'message' must be of type string}}
              details:
                type: integer \# Noncompliant {{OAR029: 'details' must be of type array}}
              httpStatus:
                type: string \# Noncompliant {{OAR029: 'httpStatus' must be of type integer}}
            required:
              - code
              - message
              - httpStatus
  

Compliant Solution (OpenAPI 3)
------------------------------

openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /endpoint:
    get:
      responses:
        200:
          description: Ok
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/response'
        204:
          description: No content
        400:
          description: Error
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/response'
        default:
          description: Unknown
components:
  schemas:
    response:
      type: object
      allOf:
        - $ref: '#/components/schemas/standard\_response'
      properties:
        data:
          type: object
          properties:
            tipos:
              type: array
              items:
                type: string
      required:
        - data
        - status
    standard\_response:
      type: object
      properties:
        status:
          type: object
          properties:
            http\_status:
              type: string
            code:
              type: integer
            errors:
              type: array
              items:
                type: object
                properties: 
                  name:
                    type: string
                  value:
                    type: string
            description:
              type: string
            internal\_code:
              type: string
          required:
            - http\_status
            - code
            - description
            - errors