When reading a URL, it must be possible to interpret which resources it is referring to in a concrete way

Ambiguous names in resources must be avoided. The recommendation is to use domain specific names.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /elements: \# Noncompliant {{OAR032: Ambiguous path parts not encouraged: elements}}
      get:
        responses:
          200:
            description: Ok
    /a/nested/items: \# Noncompliant {{OAR032: Ambiguous path parts not encouraged: items}}
      get:
        responses:
          200:
            description: Ok
    /another/{param}/with/valores: \# Noncompliant {{OAR032: Ambiguous path parts not encouraged: valores}}
      get:
        responses:
          200:
            description: Ok

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /invoices/{id}/products:
    get:
      parameters:
        - name: id
          in: path
          required: true
          type: integer
      responses:
        200:
          description: Ok

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /invoices/{id}/elements: \# Noncompliant {{OAR032: Ambiguous path parts not encouraged: elements}}
    get:
      parameters:
      - name: id
        in: path
        required: true
        schema:
          type: integer
      responses:
        200:
          description: Ok

Compliant Solution (OpenAPI 3)
------------------------------

openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /invoices/{id}/products:
    get:
      parameters:
      - name: id
        in: path
        required: true
        schema:
          type: integer
      responses:
        200:
          description: Ok