Some parameters should not pass through some querystring

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

### In case email and password are forbidden

  swagger: '2.0'
  info:
    version: 1.0.0
    title: My API
  paths:
    /examples:
      get:
        summary: Get a list of items
        parameters: \# Noncompliant {{OAR083: The parameter email should not pass through this querystring}}
          - name: email
            in: query
            type: string
          - name: param3
            in: query
            type: boolean
        responses:
          200:
            description: A list of items
            schema:
              type: array
              items:
                type: string

Compliant Solution (OpenAPI 2)
------------------------------

  swagger: '2.0'
  info:
    version: 1.0.0
    title: My API
  paths:
    /examples:
      get:
        summary: Get a list of items
        parameters:
          - name: param1
            in: query
            type: string
          - name: param2
            in: query
            type: integer
        responses:
          200:
            description: A list of items
            schema:
              type: array
              items:
                type: string

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

  openapi: "3.0.0"
  info:
    version: 1.0.0
    title: My API
  paths:
    /examples:
      get:
        summary: Get a list of items
        parameters: \# Noncompliant {{OAR083: The parameter email should not pass through this querystring}}
          - name: email
            in: query
            schema:
              type: string
        responses:
          200:
            description: A list of items
            content:
              application/json:
                schema:
                  type: array
                  items:
                    type: string

Compliant Solution (OpenAPI 3)
------------------------------

  openapi: "3.0.0"
  info:
    version: 1.0.0
    title: My API
  paths:
    /examples:
      get:
        summary: Get a list of items
        parameters:
          - name: param1
            in: query
            schema:
              type: string
          - name: param2
            in: query
            schema:
              type: integer
        responses:
          200:
            description: A list of items
            content:
              application/json:
                schema:
                  type: array
                  items:
                    type: string