All API operations must define security or inherit it from the global configuration.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

``` yaml
swagger: "2.0"
info:
  version: "1.0.0"
  title: "Sample API"
paths:
  /users:
    get: # Noncompliant {{OAR078: All API operations must define security or inherit it from the global configuration}}
      responses:
        200:
          description: "OK"
```

Compliant Solution (OpenAPI 2)
------------------------------

``` yaml
swagger: "2.0"
info:
  version: "1.0.0"
  title: Sample API
paths:
  /users:
    get:
      security:
        - apiKey: []
      responses:
        "200":
          description: OK
```

Compliant Solution - Global security (OpenAPI 2)
------------------------------

``` yaml
swagger: "2.0"
info:
  version: "1.0.0"
  title: Sample API
security:
  - apiKey: []
paths:
  /users:
    get:
      responses:
        "200":
          description: OK
```

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

``` yaml
openapi: 3.0.0
info:
  version: 1.0.0
  title: Sample API
paths:
  /users:
    get: # Noncompliant {{OAR078: All API operations must define security or inherit it from the global configuration}}
      responses:
        '200':
          description: OK
```

Compliant Solution (OpenAPI 3)
------------------------------

``` yaml
openapi: 3.0.0
info:
  version: 1.0.0
  title: Sample API
paths:
  /users:
    get:
      security:
        - apiKey: []
      responses:
        '200':
          description: OK
```

Compliant Solution - Global security (OpenAPI 3)
------------------------------

``` yaml
openapi: 3.0.0
info:
  version: 1.0.0
  title: Sample API
security:
  - apiKey: []
paths:
  /users:
    get:
      responses:
        '200':
          description: OK
```