An API definition with RBAC must take into account security responses.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  securityDefinitions:
    BasicAuth:
      type: basic
    ApiKeyAuth:
      type: apiKey
      in: header
      name: X-API-Key
    OAuth2:
      type: oauth2
      flow: accessCode
      authorizationUrl: https://example.com/oauth/authorize
      tokenUrl: https://example.com/oauth/token
      scopes:
        read: Grants read access
        write: Grants write access
        admin: Grants read and write access to administrative information
  paths:
    /with-auth-and-header:
      get:
        parameters:
          - in: header
            name: x-api-key
            type: string
        security:
          - ApiKeyAuth: \[\]
          - OAuth2: \[read, write\]
        responses: \# Noncompliant {{OAR035: Response code 401 must be defined for operations with security schemes defined}} {{OAR035: Response code 403 must be defined for operations with security schemes defined}}
          200:
            description: Ok
    /with-header:
      get:
        parameters:
          - in: header
            name: x-api-key
            type: string
        responses:
          200:
            description: Ok
    /without:
      get:
        responses:
          200:
            description: Ok

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
securityDefinitions:
  ApiKeyAuth:
    type: apiKey
    in: header
    name: X-API-Key
  OAuth2:
    type: oauth2
    flow: accessCode
    authorizationUrl: https://example.com/oauth/authorize
    tokenUrl: https://example.com/oauth/token
    scopes:
      read: Grants read access
      write: Grants write access
      admin: Grants read and write access to administrative information
paths:
  /with-auth-and-header:
    get:
      parameters:
        - in: header
          name: x-api-key
          type: string
      security:
        - ApiKeyAuth: \[\]
        - OAuth2: \[read, write\]
      responses:
        200:
          description: Ok
        401:
          description: Unauthorized
        403:
          description: Forbidden
        429:
          description: Forbidden
  /with-header:
    get:
      parameters:
        - in: header
          name: x-api-key
          type: string
      responses:
        200:
          description: Ok
        401:
          description: Unauthorized
        429:
          description: Unauthorized
  /without:
    get:
      responses:
        200:
          description: Ok

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

  openapi: "3.0.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  components:
    securitySchemes:
      BasicAuth:
        type: http
        scheme: basic
      BearerAuth:
        type: http
        scheme: bearer
      ApiKeyAuth:
        type: apiKey
        in: header
        name: X-API-Key
      OpenID:
        type: openIdConnect
        openIdConnectUrl: https://example.com/.well-known/openid-configuration
      OAuth2:
        type: oauth2
        flows:
          authorizationCode:
            authorizationUrl: https://example.com/oauth/authorize
            tokenUrl: https://example.com/oauth/token
            scopes:
              read: Grants read access
              write: Grants write access
              admin: Grants access to admin operations
  security:
    - ApiKeyAuth: \[\]
    - OAuth2:
      - read
      - write
  paths:
    /with-auth-and-header:
      get:
        parameters:
          - in: header
            name: x-api-key
            schema:
              type: string
        responses: \# Noncompliant {{OAR035: Response code 401 must be defined for operations with security schemes defined}} {{OAR035: Response code 403 must be defined for operations with security schemes defined}}
          200:
            description: Ok
  
    /with-header:
      parameters:
        - in: header
          name: x-api-key
          schema:
            type: string
      get:
        responses: \# Noncompliant {{OAR035: Response code 401 must be defined for operations with security schemes defined}} {{OAR035: Response code 403 must be defined for operations with security schemes defined}} 
          200:
            description: Ok
  
    /without:
      get:
        responses: \# Noncompliant {{OAR035: Response code 401 must be defined for operations with security schemes defined}} {{OAR035: Response code 403 must be defined for operations with security schemes defined}}
          200:
            description: Ok

Compliant Solution (OpenAPI 3)
------------------------------

openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
components:
  securitySchemes:
    ApiKeyAuth:
      type: apiKey
      in: header
      name: X-API-Key
    OAuth2:
      type: oauth2
      flows:
        authorizationCode:
          authorizationUrl: https://example.com/oauth/authorize
          tokenUrl: https://example.com/oauth/token
          scopes:
            read: Grants read access
            write: Grants write access
            admin: Grants access to admin operations
paths:
  /with-auth-and-header:
    get:
      parameters:
        - in: header
          name: x-api-key
          schema:
            type: string
      security:
        - ApiKeyAuth: \[\]
        - OAuth2:
          - read
          - write
      responses:
        200:
          description: Ok
        401:
          description: Unauthorized
        403:
          description: Forbidden
        429:
          description: Forbidden
  /with-header:
    parameters:
      - in: header
        name: x-api-key
        schema:
          type: string
    get:
      responses:
        200:
          description: Ok
        401:
          description: Unauthorized
        429:
          description: Unauthorized
  /without:
    get:
      responses:
        200:
          description: Ok