The examples can help developers to understand the response data structure and representation.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /pets:
    get:
      responses:
        206: \# Noncompliant {{OAR031: Responses must have one or more examples defined}}
          description: Pet list
          schema:
            $ref: '#/definitions/pets'
definitions:
  pet:
    type: object
    properties:
      name: \# Noncompliant {{OAR031: Properties must have an example defined}}
        type: string
      type: \# Noncompliant {{OAR031: Properties must have an example defined}}
        type: string
  pets:
    type: object
    properties:
      size: \# Noncompliant {{OAR031: Properties must have an example defined}}
        type: integer
      pets:
        type: array
        items:
          $ref: '#/definitions/pet'

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /pets:
    get:
      responses:
        206:
          description: Pet list
          schema:
            $ref: '#/definitions/pets'
          examples:
            application/json:
              size: 2
              pets:
                - name: Fluffy
                  type: cat
                - name: Sparky
                  type: dog

definitions:
  pet:
    type: object
    properties:
      name:
        type: string
        example: "Snow"
      type:
        type: string
        example: "dog"
  pets:
    type: object
    properties:
      size:
        type: integer
        example: 1
      pets:
        type: array
        items:
          $ref: '#/definitions/pet'

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /pets:
    get:
      responses:
        206: \# Noncompliant {{OAR031: Responses must have one or more examples defined}}
          description: Pet list
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/pets'
components:
  schemas:
    pet:
      type: object
      properties:
        name: \# Noncompliant {{OAR031: Properties must have an example defined}}
          type: string
        type: \# Noncompliant {{OAR031: Properties must have an example defined}}
          type: string
    pets:
      type: object
      properties:
        size: \# Noncompliant {{OAR031: Properties must have an example defined}}
          type: integer
        pets:
          type: array
          items:
            $ref: '#/components/schemas/pet'

Compliant Solution (OpenAPI 3)
------------------------------

openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /pets:
    get:
      responses:
        206:
          description: Pet list
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/pets'
              example:
                size: 2
                pets:
                - name: Fluffy
                  type: cat
                - name: Sparky
                  type: dog
components:
  schemas:
    pet:
      type: object
      properties:
        name:
          type: string
          example: Snow
        type:
          type: string
          example: dog
    pets:
      type: object
      properties:
        size:
          type: integer
          example: 1
        pets:
          type: array
          items:
            $ref: '#/components/schemas/pet'