The verbs allowed by the standard are: POST, GET, PUT, PATCH and DELETE. The rest of the verbs, even if they are allowed by REST, should not be used.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    post:
      responses: 
        200:
          description: OK
    get:
      responses: 
        200:
          description: OK
    put:
      responses: 
        200:
          description: OK
    patch:
      responses: 
        200:
          description: OK
    delete:
      responses: 
        200:
          description: OK
    head: \# Noncompliant {{OAR008: Http verb (head) not encouraged}}
      responses: 
        200:
          description: OK

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    post:
      responses: 
        200:
          description: OK
    get:
      responses: 
        200:
          description: OK
    put:
      responses: 
        200:
          description: OK
    patch:
      responses: 
        200:
          description: OK
    delete:
      responses: 
        200:
          description: OK

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: "3.0.0"
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    post:
      responses: 
        200:
          description: OK
    get:
      responses: 
        200:
          description: OK
    put:
      responses: 
        200:
          description: OK
    patch:
      responses: 
        200:
          description: OK
    delete:
      responses: 
        200:
          description: OK
    head: \# Noncompliant {{OAR008: Http verb (head) not encouraged}}
      responses: 
        200:
          description: OK

Compliant Solution (OpenAPI 3)
------------------------------

openapi: "3.0.0"
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    post:
      responses: 
        200:
          description: OK
    get:
      responses: 
        200:
          description: OK
    put:
      responses: 
        200:
          description: OK
    patch:
      responses: 
        200:
          description: OK
    delete:
      responses: 
        200:
          description: OK