Response codes in PATCH operations must be defined according to the standard

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /status:
      patch:
        summary: Get API status
        responses:
          400:
            description: error
    /another:
      patch:
        summary: Get another API status
        responses:
          400:
            description: error
    /users:
      patch:
        summary: Get all users
        responses: \# Noncompliant {{OAR064: Response code 200, 202, 204, 206 must be defined}}
          400:
            description: Error
         

Compliant Solution (OpenAPI 2)
------------------------------

  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /status:
      patch:
        summary: Get API status
        responses:
          200:
            description: OK
          202:
            description: Accepted
          206:
            description: Partial Content
    /users:
      patch:
        summary: Get all users
        responses:
          200:
            description: OK
          202:
            description: Accepted
          206:
            description: Partial Content
  

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

  openapi: "3.0.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /status:
      patch:
        summary: Get API status
        responses:
          400:
            description: error
    /another:
      patch:
        summary: Get another API status
        responses:
          400:
            description: error
    /users:
      patch:
        summary: Get all users
        responses:  \# Noncompliant {{OAR064: Response code 200, 202, 204, 206 must be defined}}
          400:
            description: Error 

Compliant Solution (OpenAPI 3)
------------------------------

  openapi: "3.0.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /status:
      patch:
        summary: Get API status
        responses:
          200:
            description: OK
          202:
            description: Accepted
          206:
            description: Partial Content
    /users:
      patch:
        summary: Get all users
        responses:
          200:
            description: OK
          202:
            description: Accepted
          206:
            description: Partial Content