For security reasons and as a REST best practice, the HTTPS protocol is mandatory.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  schemes: \# Noncompliant {{OAR001: Protocol https is mandatory}}
    - http
  paths:
    /pets: {}

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
schemes:
  - https
paths: {}

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

  openapi: "3.0.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  servers:
    - url: http://petstore.swagger.io/v1 \# Noncompliant {{OAR001: Protocol https is mandatory}}
  paths:
    /pets: {}
  

Compliant Solution (OpenAPI 3)
------------------------------

openapi: "3.0.0"
info:
  title: Swagger Petstore
  version: "1.0"
servers:
  - url: https://petstore.swagger.io/
paths: {}