Numeric properties must define a `format` attribute when their type is numeric.
This rule ensures that properties with type `integer` or `number` explicitly declare a format.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

In this example, numeric properties are missing the required `format` attribute.

``` yaml
swagger: '2.0'
info:
  version: 1.0.0
  title: My API
paths:
  /examples:
    get:
      summary: Get a list of items
      responses:
        200:
          description: A list of items
          schema:
            type: object
            properties:
              age: # Noncompliant {{OAR016: Invalid format for integer type}}
                type: integer
                format: big
              price: # Noncompliant {{OAR016: Invalid format for number type}}
                type: number
                format: decimal
```

Compliant Solution (OpenAPI 2)
------------------------------

Numeric properties correctly define a `format` attribute.

``` yaml
swagger: '2.0'
info:
  version: 1.0.0
  title: My API
paths:
  /examples:
    get:
      summary: Get a list of items
      responses:
        200:
          description: A list of items
          schema:
            type: object
            properties:
              age:
                type: integer
                format: int32
              quantity:
                type: integer
              price:
                type: number
                format: double
```

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

In this example, numeric properties do not declare a `format`.

``` yaml
openapi: "3.0.0"
info:
  version: 1.0.0
  title: My API
paths:
  /examples:
    get:
      summary: Get a list of items
      responses:
        200:
          description: A list of items
          content:
            application/json:
              schema:
                type: object
                properties:
                  quantity: # Noncompliant {{OAR016: Invalid format for integer type}}
                    type: integer
                    format: large
                  total: # Noncompliant {{OAR016: Invalid format for number type}}
                    type: number
                    format: decimal
```

Compliant Solution (OpenAPI 3)
------------------------------

Numeric properties define a `format` attribute as required.

``` yaml
openapi: "3.0.0"
info:
  version: 1.0.0
  title: My API
paths:
  /examples:
    get:
      summary: Get a list of items
      responses:
        200:
          description: A list of items
          content:
            application/json:
              schema:
                type: object
                properties:
                  quantity:
                    type: integer
                    format: int64
                  count:
                    type: integer
                  total:
                    type: number
                    format: float
```