APIs must define at most one body parameter per operation. In OpenAPI 2, an operation can only contain a single parameter defined with "in: body". Defining more than one body parameter in the same operation is not allowed and represents a design error.

This rule applies to OpenAPI 2 only.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

```yaml
swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /pets:
    post:
      parameters:
        - in: body
          name: someParam
          required: true
          schema:
            type: string
        - in: body
          name: otherParam
          required: true
          schema:
            type: string
      responses:
        default:
          description: the default response
```

Compliant Solution (OpenAPI 2)
------------------------------

```yaml
swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /pets:
    post:
      parameters:
        - in: body
          name: someParam
          required: true
          schema:
            type: string
        - in: query
          name: otherParam
          required: true
          schema:
            type: string
      responses:
        default:
          description: the default response
```

OpenAPI 3 Note
--------------

This rule does not apply to OpenAPI 3.