The last part of the path should be the API version

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
basePath: /api-seguros/v1.0 \# Noncompliant {{OAR100: Last path part must be the API version, indicated with the prefix 'v' and the version number as integer}}
paths: {}

Compliant Solution (OpenAPI 2)
------------------------------

swagger: "2.0"
info:
    version: 1.0.0
    title: Swagger Petstore
basePath: /api-seguros/v1
paths: {}

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
servers:
  - url: https://petstore.swagger.io/api-seguros/v1.0 \# Noncompliant {{OAR100: Last path part must be the API version, indicated with the prefix 'v' and the version number as integer}}
paths: {}

Compliant Solution (OpenAPI 3)
------------------------------

openapi: "3.0.0"
info:
    version: 1.0.0
    title: Swagger Petstore
servers:
    - url: https://petstore.swagger.io/api-seguros/v1
paths: {}