The schemas should match the provided examples

Noncompliant Code Example (OpenAPI 2)
---------------------------------------------------------------------------------------

    swagger: '2.0'
    info:
      title: Ejemplo de API
      version: 1.0.0
    paths:
      /item:
        get:
          responses:
            200:
              description: OK
              schema:
                type: object
                properties:
                  id:
                    type: integer
                  nombre:
                    type: string
                required:
                  - id
                  - nombre
              examples: # Noncompliant {{OAR108: Schema does not match the provided example}}
                application/json:
                  id: "123" 
                  nombre: "Ejemplo"

Compliant Solution (OpenAPI 2)
---------------------------------------------------------------------------------------

    swagger: '2.0'
    info:
      title: Ejemplo de API
      version: 1.0.0
    paths:
      /item:
        get:
          responses:
            200:
              description: OK
              schema:
                type: object
                properties:
                  id:
                    type: integer
                  nombre:
                    type: string
                required:
                  - id
                  - nombre
              examples:
                application/json:
                  id: 123
                  nombre: "Ejemplo"

Noncompliant Code Example (OpenAPI 3)
---------------------------------------------------------------------------------------

    openapi: 3.0.0
    info:
      title: Ejemplo de API
      version: 1.0.0
    paths:
      /item:
        get:
          responses:
            '200':
              description: OK
              content:
                application/json:
                  schema:
                    type: object
                    properties:
                      id:
                        type: integer
                      nombre:
                        type: string
                    required:
                      - id
                      - nombre
                  example: # Noncompliant {{OAR108: Schema does not match the provided example}}
                    id: "123"  
                    nombre: Ejemplo

Compliant Solution (OpenAPI 3)
---------------------------------------------------------------------------------------

    openapi: 3.0.0
    info:
      title: Ejemplo de API
      version: 1.0.0
    paths:
      /item:
        get:
          responses:
            '200':
              description: OK
              content:
                application/json:
                  schema:
                    type: object
                    properties:
                      id:
                        type: integer
                      nombre:
                        type: string
                    required:
                      - id
                      - nombre
                  example:
                    id: 123
                    nombre: Ejemplo