Response codes 5XX should not be defined, instead use default

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

  swagger: '2.0'
  info:
    title: Ejemplo de API con 500
    version: 1.0.0
  host: ejemplo.com
  schemes:
    - https
  paths:
    /ejemplo:
      get:
        summary: Ejemplo de operación
        responses: \# Noncompliant {{OAR109: Use default instead of directly specifying 5XX codes}}
          200:
            description: Éxito
          500:
            description: Error interno del servidor
            schema:
              $ref: '#/definitions/Error'

Compliant Solution (OpenAPI 2)
------------------------------

    swagger: "2.0"
    info:
      version: "1.0.0"
      title: "Swagger Petstore"
    host: "example.com"
    basePath: "/"
    schemes:
      - "https"
    consumes:
      - "application/json"
    produces:
      - "application/json"
    paths:
      /status:
        delete:
          summary: "Get API status"
          responses:
            "400":
              description: "error"
      /another:
        delete:
          summary: "Get another API status"
          responses:
            "400":
              description: "error"
      /users:
        delete: 
          summary: "Get all users"
          responses: 
            "400":
              description: "Error"

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

  openapi: 3.0.0
  info:
    title: Ejemplo de API con 500
    version: 1.0.0
  paths:
    /ejemplo:
      get:
        summary: Ejemplo de operación
        responses: \# Noncompliant {{OAR109: Use default instead of directly specifying 5XX codes}}
          '200':
            description: Éxito
          '500':
            description: Error interno del servidor
            content:
              application/json:
                schema:
                  $ref: '#/components/schemas/Error'

Compliant Solution (OpenAPI 3)
------------------------------

  openapi: 3.0.0
  info:
    title: Ejemplo de API con Default
    version: 1.0.0
  paths:
    /ejemplo:
      get:
        summary: Ejemplo de operación
        responses:
          '200':
            description: Éxito
          default:
            description: Error inesperado en el servidor
            content:
              application/json:
                schema:
                  $ref: '#/components/schemas/Error'