in the verb Get, some path structures are not encouraged. This can be a hint of a design flaw.

Forbidden paths by default
--------------------------

VERB

STRUCTURE

GET

/resources/get

GET

/resources/delete

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /resources/get:
    get: \# Noncompliant {{OAR103: Operation not recommended for resource path: GET /resources/get}}
      parameters:
      - name: customerId
        in: path
        required: true
        type: integer
      - name: body
        in: body
        required: true
        schema:
          type: object
      responses:
        200:
          description: Ok

Compliant Solution (OpenAPI 2)
------------------------------

    swagger: "2.0"
    info:
      title: Swagger Petstore
      version: 1.0.0
    paths:
      /resources:
        get: 
          parameters:
          - name: customerId
            in: path
            required: true
            type: integer
          - name: body
            in: body
            required: true
            schema:
              type: object
          responses:
            200:
              description: Ok

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
servers:
- url: /
paths:
  /resources/delete:
    get: \# Noncompliant {{OAR103: Operation not recommended for resource path: GET /resources/delete}}
      parameters:
      - name: customerId
        in: path
        required: true
        schema:
          type: integer
      requestBody:
        content:
          application/json:
            schema:
              type: object
        required: true
      responses:
        200:
          description: Ok

Compliant Solution (OpenAPI 3)
------------------------------

openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
servers:
- url: /
paths:
  /resources/{r\_id}/other:
    get:
      responses:
        '200':
          description: Ok