License information cannot be empty

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

    swagger: "2.0"
    info:
      title: API de Ejemplo
      version: "1.0.0"
      description: Esta es una API de ejemplo.
      termsOfService: "http://ejemplo.com/terminos/"
      contact:
        name: Soporte de API
        url: "http://ejemplo.com/contacto"
        email: "contacto@ejemplo.com"
      license: \# Noncompliant {{OAR110: License information cannot be empty}}
    host: "api.ejemplo.com"
    schemes:
      - https
    paths: {}

Compliant Solution (OpenAPI 2)
------------------------------

    swagger: "2.0"
    info:
      title: API de Ejemplo
      version: "1.0.0"
      description: Esta es una API de ejemplo.
      termsOfService: "http://ejemplo.com/terminos/"
      contact:
        name: Soporte de API
        url: "http://ejemplo.com/contacto"
        email: "contacto@ejemplo.com"
      license:
        name: "Apache 2.0"
        url: "https://www.apache.org/licenses/LICENSE-2.0.html"
    host: "api.ejemplo.com"
    schemes:
      - https
    paths: {}

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

    openapi: 3.0.0
    info:
      title: API de Ejemplo
      version: 1.0.0
      description: Esta es una API de ejemplo.
      termsOfService: http://ejemplo.com/terminos/
      contact:
        name: Soporte de API
        url: http://ejemplo.com/contacto
        email: contacto@ejemplo.com
      license: \# Noncompliant {{OAR110: License information cannot be empty}}
    paths: {}

Compliant Solution (OpenAPI 3)
------------------------------

    openapi: 3.0.0
    info:
      title: API de Ejemplo
      version: 1.0.0
      description: Esta es una API de ejemplo.
      termsOfService: http://ejemplo.com/terminos/
      contact:
        name: Soporte de API
        url: http://ejemplo.com/contacto
        email: contacto@ejemplo.com
      license:
        name: Apache 2.0
        url: https://www.apache.org/licenses/LICENSE-2.0.html
    paths: {}