Field or extension must be at the assigned location.
This rule validates that a specific field or extension is defined only at the configured locations of the OpenAPI document.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

``` yaml
swagger: "2.0"
info:
  version: "1.0.0"
  title: "Sample API"
paths:
  /example:
    get: # Noncompliant {{OAR113: Field or extension x-custom-example must be at the assigned location}}
      description: "Get example"
      responses:
        "200":
          description: "Successful response"
          content:
            application/json:
              schema:
                type: object
```

Compliant Solution (OpenAPI 2)
------------------------------

``` yaml
swagger: "2.0"
info:
  version: "1.0.0"
  title: "Sample API"
paths:
  /example:
    x-custom-example: "example value"
    get:
      x-custom-example: "example value"
      description: "Get example"
      responses:
        "200":
          x-custom-example: "example value"
          description: "Successful response"
          schema:
            type: object
```

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

``` yaml
openapi: "3.0.0"
info:
  version: "1.0.0"
  title: "Sample API"
paths:
  /example:
    get: # Noncompliant {{OAR113: Field or extension x-custom-example must be at the assigned location}}
      description: "Get example"
      responses:
        "200":
          description: "Successful response"
          content:
            application/json:
              schema:
                type: object
```

Compliant Solution (OpenAPI 3)
------------------------------

``` yaml
openapi: "3.0.0"
info:
  version: "1.0.0"
  title: "Sample API"
paths:
  /example:
    x-custom-example: "example value"
    get:
      x-custom-example: "example value"
      description: "Get example"
      responses:
        "200":
          x-custom-example: "example value"
          description: "Successful response"
          content:
            application/json:
              schema:
                type: object
```