in the verb Post, some path structures are not encouraged. This can be a hint of a design flaw.

All paths that end in a parameter are forbidden
-----------------------------------------------

Forbidden paths by default
--------------------------

VERB

STRUCTURE

POST

/resources/me

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /resources/{r\_id}:
    post: \# Noncompliant {{OAR104: Operation not recommended for resource path: resources/{r\_id}}}
      parameters:
        - name: r\_id
          in: path
          required: true
          type: string
      responses:
        200:
          description: Ok

Compliant Solution (OpenAPI 2)
------------------------------

    swagger: "2.0"
    info:
      title: Swagger Petstore
      version: 1.0.0
    paths:
        /resources/{r\_id}/other:
            post:
                parameters:
                    - name: r\_id
                    in: path
                    required: true
                    type: string
                responses:
                    200:
                    description: Ok

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
servers:
- url: /
paths:
  /resources/me: \# Noncompliant {{OAR104: Operation not recommended for resource path: resources/me}}
    post:
      responses:
        '200':
          description: Ok

Compliant Solution (OpenAPI 3)
------------------------------

openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
servers:
- url: /
paths:
    /resources/{r\_id}/other:
        post:
            responses:
                '200':
                description: Ok