in the verb Delete, some path structures are not encouraged. This can be a hint of a design flaw.

Forbidden paths by default
--------------------------

VERB

STRUCTURE

DELETE

/resources/get

DELETE

/resources/delet

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /resources/get:
    delete: \# Noncompliant {{OAR107: Operation not recommended for resource path: DELETE /resources/get}}
      responses:
        '200':
          description: Ok

Compliant Solution (OpenAPI 2)
------------------------------

    swagger: "2.0"
    info:
      title: Swagger Petstore
      version: 1.0.0
    paths:
      /resources/{r\_id}:
        delete: 
          responses:
            '200':
              description: Ok

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
servers:
- url: /
paths:
  /resources/get:
    delete: \# Noncompliant {{OAR107: Operation not recommended for resource path: DELETE /resources/get}}
      responses:
        '200':
          description: Ok

Compliant Solution (OpenAPI 3)
------------------------------

openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
servers:
- url: /
paths:
  /resources/me: 
    delete: 
      responses:
        '200':
          description: Ok