The first part of the path should be one of the alloweds

In case 'hello' is the only allowed

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
    version: 1.0.0
    title: Swagger Petstore
basePath: /comercio/balances # Noncompliant {{OAR101: The first part of the path should be one of the alloweds}}
paths: {}

Compliant Solution (OpenAPI 2)
-------------------------------------

swagger: "2.0"
info:
    version: 1.0.0
    title: Swagger Petstore
basePath: /hello/balances 
paths: {}

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
servers:
  - url: https://domain.com/entregas/balances # Noncompliant {{OAR101: The first part of the path should be one of the alloweds}}
paths: {}

Compliant Solution (OpenAPI 3)
-------------------------------------

openapi: "3.0.0"
info:
    version: 1.0.0
    title: Swagger Petstore
servers:
    - url: https://domain.com/hello/balances 
paths: {}