The data in the required field must exist in schema parameters.
All fields listed in the required array must be explicitly defined in the schema properties.

Noncompliant Code Example (OpenAPI 2)
-------------------------------------

``` yaml
swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0.0"
paths: {}
definitions:
  ErrorResponse:
    type: object
    properties:
      code:
        type: integer
      message:
        type: string
    required:
      - code
      - message
      - otherfield # Noncompliant {{OAR115: This value does not exist, it must be defined in the schema properties}}
```

Compliant Solution (OpenAPI 2)
------------------------------

``` yaml
swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0.0"
paths: {}
definitions:
  ErrorResponse:
    type: object
    properties:
      code:
        type: integer
      message:
        type: string
    required:
      - code
      - message
```

Noncompliant Code Example (OpenAPI 3)
-------------------------------------

``` yaml
openapi: 3.0.0
info:
  title: Swagger Petstore
  version: "1.0.0"
paths: {}
components:
  schemas:
    ErrorResponse:
      type: object
      properties:
        code:
          type: integer
        message:
          type: string
      required:
        - code
        - message
        - otherfield # Noncompliant {{OAR115: This value does not exist, it must be defined in the schema properties}}
```

Compliant Solution (OpenAPI 3)
------------------------------

``` yaml
openapi: 3.0.3
info:
  title: Swagger Petstore
  version: "1.0.0"
paths: {}
components:
  schemas:
    ErrorResponse:
      type: object
      properties:
        code:
          type: integer
        message:
          type: string
      required:
        - code
        - message
```